vtex/faststore
# feat: Adds B2B Customer ID

## What's the purpose of this pull request?

As we develop more b2b features inside the `faststore`, we need to be able to identify when a customer is a `b2b` customer and if he is member of an Organization (Normally refered to as a Buyer Organization). This PR addresses the first step to do so.

For that, we added a `b2b` section to the `session` hook and mutation and inside of it, a `customerId` field, that informs us of what BuyerOrganization that customer is a part of.

Why we decided to do this inside FastStore and not as an override or extra request: The information comes from the same `sessions` api `Faststore` already uses, we just need to bring an additional field. This will not impact B2C stores, for them, it will only be an empty objetct 

## How it works?

On a B2B account that fulfills several conditions (Some specific apps, some specific versions of apps, a feature flag), the `sessions` api will return the `customerId` field on the `authentication` namespace and that will be inserted on the internal session object, like so: 

Account:  `b2bfaststoredev`: 
![image](https://github.com/user-attachments/assets/88ed84f8-0b62-406c-8596-931e09ca799d)



On a B2C account, or any account that doesn't fit all the criteria, the field will be empty and should not interfere with anything.

Account: `storeframework`
![image](https://github.com/user-attachments/assets/81dbb28c-7561-4bf9-ae56-f3a57846e9c6)


## How to test it?

To make validation easier, we have 2 previews (You will need to be logged in to see it working, as they are previews, you will need to edit the cookie): 
[B2B Preview - Session will print on console every few seconds, only on the /self-management page](https://sfj-45b4ae2--b2bfaststoredev.preview.vtex.app/self-management)

[B2C Preview - Has a button on navbar to print the session on console](https://sfj-342a275--starter.preview.vtex.app/)
The button and the `faststore/cli` dependency are the only changes here

The PRs for these previews: 
[B2B](https://github.com/vtex-sites/b2bfaststoredev.store/pull/91)
[B2C](https://github.com/vtex-sites/starter.store/pull/509)
